### ISSUE TITLE
PS Vita bugs/crashes/LUA missing

### ISSUE THREAD

**Issue by thipeto** [jtothebell/fake-08#56]
Hi. I'm testing a bunch of carts on PS Vita Pico-8 version, most of them works very well but there are some bugs yet.
Tested with **v0.0.2.4** . Created an account in here just to report these issues for you. Amazing work by the way.

## LUA ERROR/MISSING ##
3D Racing Game WIP.p8 (ERROR LOADING CART LUA)
SlipWays.p8 (ERROR LOADING CART LUA)

## BUG ##
Chimera's Garden.p8
Desert Lead 3D.p8
Explorers.p8
Hydra.p8
Jack of Spades.p8
Just One Boss.p8
Katamari Christmassy.p8
PICO ARCADE.p8
PICOWARE.p8
Pinballvania!.p8
Soulbound.p8
Storming The Grandmothership.p8
Sugar Rush.p8
Super Contra Hot.p8
Super Pico Runner Brothers.p8
Tomato Worm.p8

## CRASH ##
Across The River.p8
Crowded Dungeon Crawler.p8
Delia Mute in Grave Grotto.p8
Demon Castle.p8
Gifts from Venus.p8
GRiPPY.p8
Impossible Mission R.T..p8
Lemmtris.p8
Lych King.p8
Masters of the Universe.p8
PicoKart.p8
Picopolis.p8
Ramps.p8
Scrap Boy.p8
Sonic 2.5 SAGE 2020.p8
The Adventures of Jelpi.p8
The Chamber Scrolls.p8
The Stroll.p8
Tower Noire.p8
Willo.p8

**Comment by jtothebell** [comment]
Thanks for the report! I know at least Sonic and Jelpi crashes will be fixed in the next release. I will take a look at the rest of these carts as I have time. I may have to check back here if I need more specifics about the problem.

**Comment by thipeto** [comment]
No worries dude take your time, no rushing. Just decided to improve my feedback. Good luck!

## LUA ERROR/MISSING ##
3D Racing Game WIP.p8 | work in progress (https://www.lexaloffle.com/bbs/?pid=16351#p)
SlipWays.p8 | (https://www.lexaloffle.com/bbs/?pid=50626#p)
Chimeras Garden.p8 | work in progress (https://www.lexaloffle.com/bbs/?tid=27877)

## GENERAL BUGS ##
Desert Lead 3D.p8 | (https://www.lexaloffle.com/bbs/?pid=43451)
Issue: buttons does't work

Explorers.p8 | (https://www.lexaloffle.com/bbs/?pid=82024#p)
issue: black screen

Hydra.p8 | (https://www.lexaloffle.com/bbs/?pid=19600#p)
issue: split screen in half and slow speed

Jack of Spades.p8 | (https://www.lexaloffle.com/bbs/?pid=65294#p)
issue: screen frozen

Katamari Christmassy.p8 | (https://www.lexaloffle.com/bbs/?pid=71186#p)
issue: messy controllers

PICO ARCADE.p8 | (https://www.lexaloffle.com/bbs/?pid=72470#p)
issue: unable to launch games with "X"

PICOWARE.p8 | (https://www.lexaloffle.com/bbs/?pid=65884#p)
issue: black screen after dialogue with mom

Pinballvania!.p8 | (https://www.lexaloffle.com/bbs/?pid=63353#p)
issue: controllers, pressing up to continue doesn't work

Soulbound.p8 | (https://www.lexaloffle.com/bbs/?pid=82570#p)
issue: difficulty to read dialogues

Storming The Grandmothership.p8 | (https://www.lexaloffle.com/bbs/?pid=89237#p)
issue: buggy intro animation, crashes after pressing Z+X to start

Sugar Rush.p8 | (https://www.lexaloffle.com/bbs/?pid=88978#p)
issue: after the counter in the beginning of the game, it appears "not so much" every time

Super Contra Hot.p8 | (https://www.lexaloffle.com/bbs/?pid=69629#p)
issue: buggy screen

Super Pico Runner Brothers.p8 | (https://www.lexaloffle.com/bbs/?pid=30095#p)
issue: black screen

Tomato Worm.p8 | (https://www.lexaloffle.com/bbs/?pid=55292#p)
issue: playable but buggy screen, shows weird screens during gameplay

## CRASHES RIGHT AFTER LAUNCHING THE CART ##
Across The River.p8 | (https://www.lexaloffle.com/bbs/?pid=10832#p)
Crowded Dungeon Crawler.p8 | (https://www.lexaloffle.com/bbs/?pid=82720#p)
Delia Mute in Grave Grotto.p8 | (https://www.lexaloffle.com/bbs/?pid=11873#p)
Demon Castle.p8 | (https://www.lexaloffle.com/bbs/?pid=66004#p)
Gifts on Venus.p8 | (https://www.lexaloffle.com/bbs/?pid=60313#p)
GRiPPY.p8 | (https://www.lexaloffle.com/bbs/?pid=86852#p)
Impossible Mission R.T..p8 | (https://www.lexaloffle.com/bbs/?pid=88937#p)
Lemmtris.p8 | (https://www.lexaloffle.com/bbs/?pid=11644#p)
Masters of the Universe.p8 | (https://www.lexaloffle.com/bbs/?pid=81037#p)
PicoKart.p8 | (https://www.lexaloffle.com/bbs/?pid=17739#p)
Picopolis.p8 | (https://www.lexaloffle.com/bbs/?pid=42185#p)
Ramps.p8 | (https://www.lexaloffle.com/bbs/?pid=77443#p)
Scrap Boy.p8 | (https://www.lexaloffle.com/bbs/?pid=82518#p)
The Chamber Scrolls.p8 | (https://www.lexaloffle.com/bbs/?pid=33420#p)
The Stroll.p8 | (https://www.lexaloffle.com/bbs/?pid=15316#p)
Tower Noire.p8 | (https://www.lexaloffle.com/bbs/?pid=11028#p)
Willo.p8 | (https://www.lexaloffle.com/bbs/?pid=83251#p)

**Comment by jtothebell** [comment]
Thanks again. I'm now in the process of going through these carts to see what the problems are. As I'm doing that, I decided to move them to a table in the wiki.The plan is to add more info as I find it, and make an issue for specific problems (and hopefully some of the issues affect multiple carts)

**Comment by thipeto** [comment]
> and hopefully some of the issues affect multiple carts

yeah, there's a good chance. Some carts were not working in version 0.2.2.3, but started to work in version 0.2.2.4, which was the period that I opened this request.

As for adding a table to the wiki do as you prefer, I'm not a developer I just opened this account to report these bugs to help you. Good luck and good work!

**Comment by jtothebell** [comment]
@thipeto A bunch more carts should be working on the latest release. You can see which ones specifically in the wiki page: https://github.com/jtothebell/fake-08/wiki/Cart-Compatibility . Thank you again!

**Comment by thipeto** [comment]
glad to be helpful

**Comment by thipeto** [comment]
Hi dude.

Just wanted to report some carts I've been testing and have issues. Most of the previous carts with errors I've tested and had no bugs anymore. Thanks for your efforts!

All tests were made on v.2.9 in June 6th 2021 - PS Vita emulator.

**GENERAL BUGS**
Pic-o-BEAT | (https://www.lexaloffle.com/bbs/?pid=89423#p)
issue: timeline doesn't showing commands

Peral - El Torpedero Submarino | (https://www.lexaloffle.com/bbs/?pid=92885#p)
issue: loop when starting a new game

Pico Night Punkin | (https://www.lexaloffle.com/bbs/?pid=91361#p)
issue: freezes screen after ENTER - soundtrack keeps playing though

**LUA ERROR OR MISSING**
Picross-8 | (https://www.lexaloffle.com/bbs/?pid=23106#p)

**Comment by jtothebell** [comment]
Thanks, I've added those carts to the wiki compatibility page, and I made a commit yesterday that sorts carts alphabetically. I've gotten started on the `load()` functionality needed for two of those carts and hope to be able to finish that up for the next release

**Comment by Meonk69-Vita** [comment]
Hi, want ro repot too, I use lastest pre-relesed version for ps vita, game like Poom (Doom clone), virtua racing demake doesnt work, it say missing lua to load, also I notice sound crackling on some game, its normal? Hopefully can be fixed, and thank you very much for released this on vita, its awesome!

**Comment by jtothebell** [comment]
Hello, thanks checking in. Virtua Racing should work in the latest pre-release version. Its not perfect, but I just double checked that it is playable. Have you downloaded the standalone version, extracted all 15 of the .p8 files into  the cart directory and loaded vracing_title.p8? If you do that, it should work (works on my machine anyway ;) ) Poom should also load into the title screen and be able to select a level if you download the standalone version. Its not really playable after that though, but hopefully it will be. Both of those should be noted on the wiki page. As far as the sound goes, yes it does pop and crackle sometimes. There are a handful of parts of the sound that aren't implemented properly, and I do plan to address it, but I'm not sure when.

**Comment by thipeto** [comment]
Hi, @jtothebell

There is a lot of carts with the "NO LUA TO LOAD. ABORTING CART". Just wanted you to know, here's the list:

https://github.com/jtothebell/fake-08/files/8354560/NO.LUA.TO.LOAD.ABORTING.CART.txt 

Tested on PS Vita with V0.0.2.15. Thanks!
